connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ConnectorConfig.java

```diff
@@ -206,7 +206,9 @@ static ConfigDef getConfigDefFromTransformation(String key, Class<?> transformat
     static final class TransformationClassRecommender implements ConfigDef.Recommender {
         @Override
         public List<Object> validValues(String name, Map<String, Object> parsedConfig) {
-            return (List) PluginDiscovery.transformationPlugins();
+            // TODO: disabled until a solid way to inject Plugins factory is selected.
+            // return (List) PluginDiscovery.transformationPlugins();
+            return Collections.emptyList();
```

Comment:
I think the old `PluginDiscovery` wasn't written with hot swapping of plugins in mind. 

Thankfully I think that's the only place where the static list was requested outside the Resources classes.